march_hardware/src/IMotionCube.cpp

```diff
@@ -35,6 +40,14 @@ IMotionCube::IMotionCube(int slave_index, std::unique_ptr<AbsoluteEncoder> absol
   // Multiply by ten to ensure the rotational joints keep using absolute encoders. These are somehow more accurate
   // even though they theoretically shouldn't be.
 }
+IMotionCube::IMotionCube(int slave_index, std::unique_ptr<AbsoluteEncoder> absolute_encoder,
+                         std::unique_ptr<IncrementalEncoder> incremental_encoder, std::string& sw_stream,
+                         ActuationMode actuation_mode)
+  : IMotionCube(slave_index, std::move(absolute_encoder), std::move(incremental_encoder), actuation_mode)
+{
+  this->sw_stream_ = std::move(sw_stream);
+  ROS_INFO("length in constructor: %lu", this->sw_stream_.length());
```

Comment:
Now that I think about it, this info message isn't very descriptive and is probably only necessary for debugging. Therefore, you should either change it to `ROS_DEBUG` and add some information, like slave index, or remove the statement altogether.